crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -477,81 +477,119 @@ private QueryExpression rewriteUpdateQueryForCompositeSearch(CompositeMetadata m
-                        if(!doc.hasErrors()) {          
-                            response.addEntityData(doc.getOutputDocument().getRoot());
-                            rmd.add(doc.getResultMetadata());
+            StreamingResponse r=_findAndStream(req,ctx);
+            
+            response = new Response(r);
+            if(response.getErrors()==null||response.getErrors().isEmpty()) {
+                DocumentStream<DocCtx> docStream=r.documentStream;
+                List<ResultMetadata> rmd=new ArrayList<>();
+                response.setEntityData(factory.getNodeFactory().arrayNode());
+                for(;docStream.hasNext();) {
+                    DocCtx doc=docStream.next();
+                    if(!doc.hasErrors()) {          
+                        response.addEntityData(doc.getOutputDocument().getRoot());
+                        rmd.add(doc.getResultMetadata());
                     } else {
-                            DataError error=doc.getDataError();
-                            if(error!=null)
-                                response.getDataErrors().add(error);
-                        }
+                        DataError error=doc.getDataError();
+                        if(error!=null)
+                            response.getDataErrors().add(error);
                     }
-                    docStream.close();
-                    response.setResultMetadata(rmd);
-                    ctx.measure.end("postProcessFound");
-                    ctx.setStatus(OperationStatus.COMPLETE);
-                } else {
-                    ctx.setStatus(OperationStatus.ERROR);
                 }
-                response.setMatchCount(result == null ? 0 : result.getSize());
+                docStream.close();
+                response.setResultMetadata(rmd);
+                response.setMatchCount(r.matchCount == null ? 0 : r.matchCount);
             }
-            // call any queued up hooks (regardless of status)
-            ctx.getHookManager().queueMediatorHooks(ctx);
-            response.setStatus(ctx.getStatus());
-            response.getErrors().addAll(ctx.getErrors());
-            if (response.getStatus() != OperationStatus.ERROR) {
-                ctx.getHookManager().callQueuedHooks();
+         } catch (Error e) {
+            LOGGER.debug("Error during find:{}", e);
+            response.getErrors().add(e);
+        } catch (Exception e) {
+            LOGGER.debug("Exception during find:{}", e);
+            response.getErrors().add(Error.get(CrudConstants.ERR_CRUD, e));
+        } finally {
+            if(ctx!=null) {
+                ctx.measure.end("find");
+                METRICS.debug("find: {}",ctx.measure);
             }
+            Error.pop();
+        }
+        
+        return response;
+    }
+    
+    @StopWatch(loggerName = "stopwatch.com.redhat.lightblue.mediator.Mediator", sizeCalculatorClass = "com.redhat.lightblue.mediator.ResponsePayloadSizeCalculator")
```

Comment:
A note on StopWatch: since findAndStream execution time will depend on the client (how fast it's consuming the results), long execution times may be fine. That's the whole point I guess, just need to keep that in mind when looking at slow queries logged.